Assets/Mirror/Runtime/NetworkWriter.cs

```diff
@@ -18,11 +18,18 @@ public class NetworkWriter
         // create writer immediately with it's own buffer so no one can mess with it and so that we can resize it.
         // note: BinaryWriter allocates too much, so we only use a MemoryStream
         readonly MemoryStream stream = new MemoryStream();
+        internal bool pooled;
+        internal readonly bool reusable;
```

Comment:
why do we need to know if the NetworkWriter was created by the pool?   

I get why we need the `pooled`,  but I don't see the value of `reusable`

Reply:
That's for other people who use our writers, this causes that their writers won't be recycled if they don't want to

Reply:
yes,  but why is that important?   so what if they recycle a NetworkWriter that they created?

Reply:
What if they have their own reusing system, and we somehow steal their writer cause it lands in our pool?

Reply:
well,  they already have to go out of their way and call `Recycle()` in order for it to appear in our pool.  I would say that is enough.

By your logic, what prevents someone from:
```
NetworkWriter writer = new NetworkWriter(true);
// recycle in a custom pool
MyPool.Recycle(writer);
NetworkWriterPool.Recycle(writer);
```

Or what prevents someone from doing?
```cs
class MyClass
{
      private static NeworkWriter writer;

      public void foo() {
            writer = NetworkWriterPool.GetPooledWriter();
            ...
            NetworkWriterPool.Recycle(writer);
     }

      public void bar() 
      {
             writer.Write("oops,  should not use this writer");
      }
}
```

The point is that they can mess it up with or without this field

Reply:
Then, that's their fault, this is checked in case they pass the writer somewhere in our code when we call Recycle

Reply:
What's the issue with having a one safety bool?

Reply:
the issue is:
1) it does not really make it safer
2) you no longer have a default constructor, which can make certain use cases more complicated, such as creating it in a generic method.
3) classes should not be aware of their life cycle or where they are used.